File: javascript_practice_problems/01_netflix/Stack.js
export class MyStack{
    constructor(){
        this.stack_list = []
    }

    is_empty(){
        return this.stack_list.length == 0
    }

    top(){
        if (this.is_empty())
            return null
        return this.stack_list[this.stack_list.length-1]
    }

    size(){
        return this.stack_list.length
    }

    push(value){
        this.stack_list.push(value)
    }

    pop(){
        if (this.is_empty())
            return null
        return this.stack_list.pop()
    }
}

File: ruby_practice_problems/02_netflix/Stack.rb
class MyStack
    # Ruby's field are always private. attr_accessor makes them public
    attr_accessor :stack_list
    def initialize()
        @stack_list = []
    end

    def is_empty()
        return @stack_list.length() == 0
    end

    def top()
        if is_empty()
            return nil
        end
        return @stack_list[-1]
    end

    def size()
        return @stack_list.length()
    end

    def push(value)
        @stack_list.push(value)
    end

    def pop()
        if is_empty()
            return nil
        end
        return @stack_list.pop()
    end
end